### PR TITLE
chore: add null placeholder for missing attributes in custom metric registry

### PR DESCRIPTION
**Pull request by sanket-mundra**
Added null placeholder value for missing attributes in entities.

### CODE REVIEW

File: document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java [hunk 1]
```diff
@@ -35,6 +41,18 @@ public List<DocStoreMetric> getCustomMetrics(final CustomMetricConfig customMetr
     try {
       final String collectionName = customMetricConfig.collectionName();
       final Query query = customMetricConfig.query();
+      query
+          .getSelections()
+          .forEach(
+              selectionSpec -> {
+                final String alias = selectionSpec.getAlias();
+                if (Objects.isNull(alias) || alias.isEmpty()) {
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java:49]:
```suggestion
                if (Objects.isNull(alias) || alias.isBlank()) {
```

File: document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java [hunk 2]
```diff
@@ -22,10 +26,12 @@
 import org.hypertrace.core.documentstore.Document;
 import org.hypertrace.core.documentstore.model.config.CustomMetricConfig;
 import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.query.SelectionSpec;
 
 @Slf4j
 @AllArgsConstructor
 public abstract class BaseDocStoreMetricProviderImpl implements DocStoreMetricProvider {
+  private static final String NULL_VALUE_PLACEHOLDER = "<null>";
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java:34]:
```suggestion
  private static final String NULL_LABEL_VALUE_PLACEHOLDER = "<null>";
```

File: document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java [hunk 3]
```diff
@@ -68,9 +86,20 @@ public List<DocStoreMetric> getCustomMetrics(final CustomMetricConfig customMetr
           continue;
         }
 
+        Map<String, JsonNode> jsonNodeMap =
+            query.getSelections().stream()
+                .collect(
+                    Collectors.toUnmodifiableMap(
+                        SelectionSpec::getAlias,
+                        selectionSpec ->
+                            Optional.ofNullable(node.get(selectionSpec.getAlias()))
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java:95]: Can you double-check if `node.get()` returns a `null` value or a `MissingNode` instance?

Comment (sanket-mundra) [reply document-store/src/main/java/org/hypertrace/core/documentstore/metric/BaseDocStoreMetricProviderImpl.java:95]: Yes it gives null, checked it

**Review by suresh-prakash — APPROVED** [review]

**Pull request merged**